compiler/rustc_trait_selection/src/traits/coherence.rs

```diff
@@ -924,9 +924,9 @@ where
                 }
             }
 
-            ty::Alias(kind @ (ty::Projection | ty::Inherent | ty::Weak), ..) => {
+            ty::Alias(kind @ (ty::Projection | ty::Inherent | ty::Weak), _) => {
                 if ty.has_type_flags(ty::TypeFlags::HAS_TY_PARAM) {
-                    bug!("unexpected ty param in alias ty");
+                    bug!("orphanck: unexpected ty param in alias ty");
                 }
 
```

Comment:
tbh, I'd just entirely remove this assert and put one at the top of `orphan_check_trait_ref` if there isn't one already